### PR TITLE
Feature gb alwaysonbot

### PR DESCRIPTION
**Pull request by rajuchetlapalli**
# Description

Always On Bot changes

### CODE REVIEW

File: .config/.env.example [hunk 1]
```diff
@@ -15,4 +15,14 @@ LuisAPIHostNameFR=canadacentral
 # Project
 PROJECT=callback-bot-backend
 ENVIRONMENT=local
-TZ=America/Vancouver
\ No newline at end of file
+TZ=America/Vancouver
+
+LuisAppIdEN=552ba77b-b2a9-4d23-b552-0d11e87e4e51
+LuisAPIKeyEN=68f69c659e7f4d3493bed91fa2ecca35
```
Comment (binggaofw) [inline .config/.env.example]: do not commit the secrets to git

**Review by binggaofw — CHANGES_REQUESTED** [review]
you delete all my previous unit test files, please add them back

File: src/dialogs/alwaysonbotDialogs/Common/continueAndFeedbackStep.ts [hunk 2]
```diff
@@ -0,0 +1,102 @@
+import { LuisRecognizer } from "botbuilder-ai";
+import {
+    Choice, ChoiceFactory, ChoicePrompt, ComponentDialog, DialogTurnResult, ListStyle, PromptValidatorContext, TextPrompt,
+    WaterfallDialog,
+    WaterfallStepContext
+} from "botbuilder-dialogs";
+import { CommonPromptValidatorModel } from "../../../models/commonPromptValidatorModel";
+import { LUISAlwaysOnBotSetup } from "../alwaysOnBotRecognizer";
+
+import i18n from "../../locales/i18nconfig";
```
Comment (binggaofw) [inline src/dialogs/alwaysonbotDialogs/Common/continueAndFeedbackStep.ts]: i18nConfig not i18nconfig

File: src/dialogs/alwaysonbotDialogs/Common/feedBackStep.ts [hunk 3]
```diff
@@ -0,0 +1,51 @@
+import {
+    Choice, ChoiceFactory, ChoicePrompt, ComponentDialog, DialogTurnResult, PromptValidatorContext, TextPrompt,
+    WaterfallDialog,
+    WaterfallStepContext
+} from "botbuilder-dialogs";
+
+import i18n from "../../locales/i18nconfig";
```
Comment (binggaofw) [inline src/dialogs/alwaysonbotDialogs/Common/feedBackStep.ts]: same here

File: src/dialogs/alwaysonbotDialogs/UpdateProfile/UpdateAddress/choiceCheckUpdateAddressStep.ts [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+import {
+    Choice, ChoiceFactory,TextPrompt,
+    ChoicePrompt, ComponentDialog, ListStyle, PromptValidatorContext, WaterfallDialog,
+    WaterfallStepContext
+} from "botbuilder-dialogs";
+import { CommonPromptValidatorModel } from "../../../../models/commonPromptValidatorModel";
+
+import { LUISAlwaysOnBotSetup } from "../../alwaysOnBotRecognizer";
+
+import i18n from "../../../locales/i18nconfig";
```
Comment (binggaofw) [inline src/dialogs/alwaysonbotDialogs/UpdateProfile/UpdateAddress/choiceCheckUpdateAddressStep.ts]: same here

File: src/dialogs/alwaysonbotDialogs/UpdateProfile/UpdateAddress/addressDetails.ts [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+    public promtMessage;
+    public promtRetryMessage;
+    constructor(){
+        this.masterError = null;
+        this.PostalCode=null;
+        this.UnitNumber=null;
+        this.FullAddress=null;
+        this.AddressType=null;
+        this.currentStep=null;
+        this.maxCount=3;
```
Comment (binggaofw) [inline src/dialogs/alwaysonbotDialogs/UpdateProfile/UpdateAddress/addressDetails.ts:24]: not sure if this should set to 2?

**Review by binggaofw — APPROVED** [review]

**Review by llakewood — APPROVED** [review]
All looks good and clean and logical to me.

File: .config/.env.example [hunk 6]
```diff
@@ -15,4 +15,14 @@ LuisAPIHostNameFR=canadacentral
 # Project
 PROJECT=callback-bot-backend
 ENVIRONMENT=local
-TZ=America/Vancouver
\ No newline at end of file
+TZ=America/Vancouver
```
Comment (llakewood) [inline .config/.env.example:18]: My only question is why Vancouver here and not Ottawa. Might make sense to set to the Central and head of state time zone by default in the example. Small change.

Comment (binggaofw) [reply .config/.env.example:18]: I do not think we are currently use TZ in our code right now. this is just an example in env.examplea as I copy it from other repo.

**Pull request merged**